test/utils/WithJwtKeys.scala

```diff
@@ -21,8 +34,28 @@ trait WithJwtKeys {
     }
   }
 
-  def generateToken(expiredIn: Int = 1000, issuedAt: Int = 0, id: String = "12345", by: String = "ego", about: String = "12345", content: String = ""): String = {
-    val claim = JwtClaim().by(by).expiresIn(expiredIn).withId(id).issuedAt(issuedAt).about(about).withContent(content)
-    JwtJson.encode(claim, privateKey, JwtAlgorithm.RS256)
+  private def stripPublicKeyText(keyText: String): String =
+    keyText
+      .stripMargin
+      .replace("\n", "")
+      .replace("-----BEGIN PUBLIC KEY-----", "")
+      .replace("-----END PUBLIC KEY-----", "")
+
+  def generateToken(expiredIn: Int = 1000, issuedAt: Int = 0, id: String = "12345", by: String = "ego", about: String = "12345", content: String = "", keyId: String = "kid1"): String = {
+    val claim = JwtClaim().by(by).expiresIn(expiredIn).withId(id).issuedAt(issuedAt).about(about).withContent(content).+("typ", "Bearer")
+    val header = JwtHeader.apply(JwtAlgorithm.RS256).withKeyId(keyId).withType("JWT")
+    JwtJson.encode(header, claim, privateKey)
+  }
+
+  def getKeycloakToken(ws: WSClient)(implicit ec: ExecutionContext): Future[String] = {
+    val params: Seq[(String, String)] = Seq(("username", "admin"), ("password", "admin"), ("grant_type", "password"), ("client_id", "admin-cli"))
+    val response: Future[WSResponse] = ws
+      .url("http://localhost:8080/auth/realms/master/protocol/openid-connect/token")
```

Comment:
I don't know the codebase but maybe you want to have hostname and url as constant somewhere outside of this function